javalib/src/main/scala/java/util/HashSet.scala

```diff
@@ -0,0 +1,96 @@
+package java.util
+
+import scala.collection.mutable
+import scala.collection.JavaConversions._
+
+class HashSet[E] extends AbstractSet[E] with Set[E]
+                                        with Cloneable
+                                        with Serializable {
+  def this(initialCapacity: Int, loadFactor: Float) =
+    this()
+
+  def this(initialCapacity: Int) =
+    this()
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  protected val inner: mutable.Map[Int, Box[E]] =
+    new mutable.HashMap[Int, Box[E]]()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(Box(o).hashCode)
```

Comment:
This still exposes the problem of hash codes. You can still have two boxes with the same hash code, but that are not equal.

Test this:
```scala
val set = new HashSet[Any]
set.add(true)
set.add(1231)
expect(set.size).toEqual(2)
```
You'll see that it breaks because `true` and `1231` have the same hash code, and you use only the hash code as the identity of the values.

The keys of the `inner` map should be `Box`es themselves, not the hash codes.
Also, why not simply use a `scala.collection.mutable.Set[Box[E]]`?